application/controllers/admin/globalsettings.php

```diff
@@ -480,6 +480,65 @@ public function surveysettingmenues()
         Yii::app()->getController()->renderPartial('super/_renderJson', ['data' => $menues[0]]);
     }
 
+    public function sendTestEmail()
+    {
+        $sSiteName = Yii::app()->getConfig('sitename');
+
+        //Use the current user details for the default administrator name and email
+        $user = User::model()->findByPk(Yii::app()->session['loginID']);
+        $sTo = $user->full_name . " <" . $user->email . ">";
+        $sFrom = Yii::app()->getConfig("siteadminname") . " <" . Yii::app()->getConfig("siteadminemail") . ">";
+        $sSubject = sprintf(gT('Test email from %s'), $sSiteName);
+        
+        $body   = array();
+        $body[] = sprintf(gT('This is a test email from %s'), $sSiteName);
+        $body   = implode("\n", $body);
+
+        $this->_sendEmailAndShowResult($body, $sSubject, $sTo, $sFrom);
+    }
+
+    private function _sendEmailAndShowResult($body, $sSubject, $sTo, $sFrom)
+    {
+        $mailer = new \LimeMailer();
+        $mailer->rawBody = $body;
+        $mailer->rawSubject = $sSubject;
+        $mailer->setTo($sTo);
+        $mailer->setFrom($sFrom);
+
+        $success = $mailer->sendMessage();
+        $maildebug = $mailer->getDebug('html');
+
+        if ($success) {
+            $content = gT('Email sent successfully');
+        } else {
+            $content = gT('Email sending failure.');
+        }
+
+        $data = [];
+        $data['message'] = $content;
+        $data['success'] = $success;
+        $data['maildebug'] = $maildebug;
+
```

Comment:
Wbhy not use : https://github.com/LimeSurvey/LimeSurvey/blob/91375a58e22334af287e652885599142954a676f/application/core/LimeMailer.php#L460-L467 ?

Reply:
You mean as to show erro details?

Reply:
This return the error from phpMailer like "can not connect" etc  … the cleanest error

Reply:
Updated